src/api-service/tests/test_task_config.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import pytest
+
+from __app__.onefuzzlib.tasks.config import is_valid_blob_name
+
+
+BLOB_NAME_TEST_CASES = [
+    # Valid
+    ("fuzz.exe", True),
+    ("bin/fuzz.exe", True),
+    ("/".join("a" * 254), True),
+    ("a" * 1024, True),
+    # Invalid (absolute)
+    ("/fuzz.exe", False),
+    ("/bin/fuzz.exe", False),
```

Comment:
can we add some test with windows style absolute path: c:\..

Reply:
I had those originally, but I think it starts getting confusing, and probably isn't where we want our API boundary.

Here's what I'm thinking.
1. The service should expect that blob names as specified by Azure (and not arbitrary, OS-dependent filesystem names). Any client-local filesystem delimiting should be normalized to a representation using only forward slashes (i.e. are POSIX-style). Any other legal characters after that forward-slash normalization should be treated as just part of the name of some path component. Note that an (escaped) backslash (`\`) is a valid part of a POSIX filename.
2. The CLI should use the client OS to help construct valid relative blob names, and should be responsible for normalizing them as POSIX-style.

The old could would actually throw exceptions if Windows absolute paths ever got to the service.

I can still add those test cases if you think it helps for documentation. But the idea is that I want to make it so the service does not have to try to detect the path delimiter style that is being used.

Reply:
I don't think the cli currently rewrites the target_exe path so it is possible to receive the windows absolute path.
Plus having the validation on the service side makes it independent from the client code

Reply:
Ok i just looked a the code and we do indeed rewrite the target_exe path. But my second point still stands.